how-to/virtualisation/sev-snp.md

```diff
@@ -0,0 +1,49 @@
+While using Ubuntu as a guest OS on SEV-SNP VMs has been supported since Ubuntu 24.04 LTS, the host enablement (QEMU and OVMF support) was only added later with Ubuntu 25.04.
+
+## Host configuration
+
+AMD SEV-SNP is fully supported on Ubuntu 25.04 and newer. To launch a VM with these features enabled, you need to make sure that SNP is enabled in the firmware settings and that Address-Space Identifiers (ASIDs) are assigned to it. You also want to make sure memory encryption features are enabled. For more details, refer to [AMD's documentation](https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/tuning-guides/58207-using-sev-with-amd-epyc-processors.pdf) and to specific server configuration manuals.
+
+On the host OS, install `qemu-system-x86_64` and launch QEMU with the following parameters:
+
+```bash
+qemu-system-x86_64 \
+  -enable-kvm \
+  -nographic \
+  -machine q35 -smp 6 -m 6G \
+  -drive "if=virtio,format=qcow2,file=disk.img" \
+  -net nic,model=e1000 -net user,hostfwd=tcp::2222-:22 \
+\
+  -cpu EPYC-v4 \
+  -machine memory-encryption=sev0,vmport=off \
+  -object memory-backend-memfd,id=ram1,size=6G,share=true,prealloc=false \
+  -machine memory-backend=ram1 \
+  -object sev-snp-guest,id=sev0,cbitpos=47,reduced-phys-bits=1,kernel-hashes=on \
+  -kernel ./vmlinuz \
+  -append "root=/dev/vda1 console=ttyS0" \
+  -bios /usr/share/ovmf/OVMF.amdsev.fd
+```
+
+Here we are configuring the VM to use 6GB of encrypted memory:
+* `cbitpos` and `reduced-phys-bits` have to be `47` and `1` respectively for all EPYC CPUs.
+* `kernel-hashes=on` makes sure the kernel, initramfs and command line will be measured during the launch. It can be disabled but if enabled, the kernel needs to be provided with `-kernel`
+* `OVMF.amdsev.fd` is a specific version of [EDK II](https://en.wikipedia.org/wiki/TianoCore_EDK_II).
+
+For more details about these parameters, refer to QEMU documentation pages for [invocation](https://www.qemu.org/docs/master/system/invocation.html) and [AMD SEV](https://www.qemu.org/docs/master/system/i386/amd-memory-encryption.html#sevapi).
+
+## Guest configuration
+
+On the guest side, Ubuntu 24.04 LTS and newer fully support AMD SEV-SNP. You can download the disk image and kernel for your VM from `cloud-images.ubuntu.com`. The latest image and kernel can be found here:
+ * https://cloud-images.ubuntu.com/releases/noble/release/ubuntu-24.04-server-cloudimg-amd64.img
+ * https://cloud-images.ubuntu.com/releases/noble/release/unpacked/
+
+Once the VM is launched, install `linux-generic` to get the `sev-guest` module and insert it with `modprobe sev-guest`. This will create a new character device on the guest that can be used to generate attestations from the SoC: `/dev/sev-guest`.
```

Comment:
'attestation' feels a bit jargon, so may be worth a short sentence explaining its meaning in this context?

Reply:
@gjolly Would you mind adding a small explanation about "attestation" to https://documentation.ubuntu.com/server/reference/glossary/ and linking to it here accordingly?

Reply:
Done, let me know if you are good with it.

Reply:
I also replaced "attestation" by "attestation report" and "SoC" by "TEE".